src/main/java/de/craften/plugins/rpgplus/components/storage/StorageComponent.java

```diff
@@ -10,17 +10,29 @@
  * A component that offers a key-value storage.
  */
 public class StorageComponent extends PluginComponentBase {
-    private final File directory;
+    private final StorageType type;
+	private final File directory;
     private Storage storage;
 
-    public StorageComponent(File directory) {
+    public StorageComponent(StorageType type, File directory) {
+    	this.type = type;
         this.directory = directory;
     }
 
     @Override
     protected void onActivated() {
         try {
-            storage = new YamlStorage(directory);
+        	switch (type) {
```

Comment:
Maybe it would be better to create three subclasses of `StorageComponent` which all override `onActivated` to do their logic instead of one class doing it all. :thinking: 
Only the `YamlStorage` would then require a directory.